frontend/lib/resolution/Resolver.cpp

```diff
@@ -4870,6 +4870,59 @@ void Resolver::exit(const Range* range) {
   }
 }
 
+bool Resolver::enter(const uast::Array* decl) {
+  return true;
+}
+void Resolver::exit(const uast::Array* decl) {
+  if (scopeResolveOnly) {
+    return;
+  }
+
+  ResolvedExpression& r = byPostorder.byAst(decl);
+
+  // Resolve call to appropriate array builder proc
+  const char* arrayBuilderProc;
+  std::vector<CallInfoActual> actuals;
+  if (!decl->isMultiDim()) {
+    arrayBuilderProc = "chpl__buildArrayExpr";
+  } else {
+    arrayBuilderProc = "chpl__buildNDArrayExpr";
+
+    // Get shape arg
+    std::vector<QualifiedType> shapeTupleElts;
+    for (auto dim : decl->shape()) {
+      shapeTupleElts.push_back(QualifiedType::makeParamInt(context, dim));
+    }
+    auto shapeTupleType = TupleType::getQualifiedTuple(context, shapeTupleElts);
+    actuals.emplace_back(
+        QualifiedType(QualifiedType::CONST_VAR, shapeTupleType),
+        UniqueString());
+  }
+
+  // Add element args
+  for (auto expr : decl->flattenedExprs()) {
+    auto exprType = byPostorder.byAst(expr).type();
+    // Short circuit if any elements have unknown type, since we won't be
+    // able to resolve the array builder proc.
+    if (exprType.isUnknown()) {
+      r.setType(QualifiedType());
+      return;
+    }
```

Comment:
In my preceding comment I recommended using `shouldSkipCallResolution` before noticing this. I see that you're avoiding unknown types, but there are a whole number of other cases (generic types without substitutions, params without known values, etc.) that should be avoided when resolving generated calls. So, I still think you should use the `shouldSkip` call, and then you can drop this case here.